src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -384,8 +393,17 @@ private void onSearchForRelationsSucceed(BibEntry entry, CheckListView<CitationR
                                              ObservableList<CitationRelationItem> observableList) {
         hideNodes(abortButton, progress);
 
-        observableList.setAll(fetchedList.stream().map(entr -> new CitationRelationItem(entr, false))
-                                         .collect(Collectors.toList()));
+        observableList.setAll(fetchedList.stream().map(entr -> {
+            Optional<BibEntry> duplicate = duplicateCheck.containsDuplicate(
+                    databaseContext.getDatabase(),
+                    entr,
+                    BibDatabaseModeDetection.inferMode(databaseContext.getDatabase()));
+
+            if (duplicate.isPresent()) {
+                return new CitationRelationItem(entr, duplicate.get(), true);
+            }
+            return new CitationRelationItem(entr, null, false);
+        }).collect(Collectors.toList()));
```

Comment:
A simple `toList()` could work - or do you need to modify the list later?

Reply:
no there is no need to modify later.